src/charm.py

```diff
@@ -71,6 +169,40 @@ def _on_config_changed(self, _):
                 value = self.model.config[key]
                 logger.info("Setting {} to: {}".format(key, value))
                 self._stored.config[key] = value
+
+        # this should go into install hook
+        uid = pwd.getpwnam("root").pw_uid
+        gid = grp.getgrnam("root").gr_gid
+
+        path = "/etc/nginx/ssl"
+        if not os.path.exists(path):
+            os.makedirs(path, 0o755)
+            os.chown(path, uid, gid)
+            os.chmod(path, 0o755)
+
+        config = self._stored.config
+        self._stored.config["ssl_enabled"] = (
+            "ssl_cert" in config.keys()  # noqa: W503
+            and "ssl_key" in config.keys()  # noqa: W503
+            and "ssl_cert" in self._stored.config.keys()  # noqa: W503
+            and "ssl_key" in self._stored.config.keys()  # noqa: W503
```

Comment:
update from "and self._stored.config["ssl_key"], the old expreesion would return a string instead of a bool